File: app/src/main/java/com/example/atthemoment/model/Category.java
package com.example.atthemoment.model;

public class Category {
    private String categoryId;
    private String categoryName;
    private boolean hasTimeTables;
    private Object icons;
    public String getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(String categoryId) {
        this.categoryId = categoryId;
    }

    public String getCategoryName() {
        return categoryName;
    }

    public void setCategoryName(String categoryName) {
        this.categoryName = categoryName;
    }

    public boolean isHasTimeTables() {
        return hasTimeTables;
    }

    public void setHasTimeTables(boolean hasTimeTables) {
        this.hasTimeTables = hasTimeTables;
    }

    public Object getIcons() {
        return icons;
    }

    public void setIcons(Object icons) {
        this.icons = icons;
    }
}

File: model/Category.swift
import Foundation

struct Category: Codable {
    var categoryId: String? 
    var categoryName: String?
    var hasTimeTables: Bool?
    var icons: [String]?

    enum CodingKeys: String, CodingKey {
        case categoryId = "CategoryId"
        case categoryName = "CategoryName"
        case hasTimeTables = "HasTimeTables"
        case icons = "Icons"
    }
}
